### PR TITLE
backEndTest

### PR DESCRIPTION
**Pull request by Ettatabe**
Implemented the snapshot method for  and added some more tests for more line coverage.

### CODE REVIEW

**Comment by TimYuenior** [comment]
Travis passes, style looks okay, I  think we can merge this.

File: src/main/java/nl/tudelft/contextproject/camera/Camera.java [hunk 1]
```diff
@@ -197,6 +197,23 @@ public void tilt(int offset) {
...
+     * Pans and Tilts the camera to a setting offset. Cannot tilt or
+     * pan the camera past the pan and tilt limits.
+     *
+     * @param panOffset The offset to pan the camera
+     * @param tiltOffset The offset to tilt the camera.
+     */
+    public void panTilt(int panOffset, int tiltOffset) {
+        camSet.panTilt(panOffset, tiltOffset);
+
+        if(hasConnection()) {
```
Comment (bvobart) [inline src/main/java/nl/tudelft/contextproject/camera/Camera.java]: Add a space between `if` and `(hasConnection()) {`

File: src/main/java/nl/tudelft/contextproject/camera/CameraConnection.java [hunk 2]
```diff
@@ -142,4 +142,9 @@
      * @return True iff the operation was performed successfully.
      */
     protected abstract boolean relFocus(int offset);
+
+    /**
+     * @param imageLocation The location to which the captured image is stored.
```
Comment (bvobart) [inline src/main/java/nl/tudelft/contextproject/camera/CameraConnection.java:150]: Missing comment on the method, even though `@param` argument does exist.

File: src/main/java/nl/tudelft/contextproject/camera/LiveCameraConnection.java [hunk 3]
```diff
@@ -1,6 +1,14 @@
 package nl.tudelft.contextproject.camera;
 
+import javafx.embed.swing.SwingFXUtils;
+import javafx.scene.SnapshotParameters;
+import javafx.scene.image.ImageView;
+import javafx.scene.image.WritableImage;
+import nl.tudelft.contextproject.gui.LiveStreamHandler;
+
+import javax.imageio.ImageIO;
 import java.io.BufferedReader;
```
Comment (bvobart) [inline src/main/java/nl/tudelft/contextproject/camera/LiveCameraConnection.java:10]: CheckStyle has some errors about these imports.

**Comment by bvobart** [comment]
I went through some CheckStyle warnings with you, which have now been fixed. Once the build passes, this branch can be merged!

**Comment by Ettatabe** [comment]
Yes. Thanks @bvobart these have been fixed.

**Comment by bvobart** [comment]
Alright, the merge passes, I think this branch can be merged!

**Comment by stevenmeijer** [comment]
Great work, merging!

**Pull request merged**